tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -330,43 +330,46 @@ def int_value_fn(idx: int) -> float:
 @pytest.mark.parametrize(
     "state",
     [
-        (optuna.trial.TrialState.FAIL,),
-        (optuna.trial.TrialState.PRUNED,),
-        (optuna.trial.TrialState.RUNNING,),
-        (optuna.trial.TrialState.WAITING,),
+        optuna.trial.TrialState.FAIL,
+        optuna.trial.TrialState.PRUNED,
+        optuna.trial.TrialState.RUNNING,
+        optuna.trial.TrialState.WAITING,
     ],
 )
 def test_sample_relative_handle_unsuccessful_states(
     state: optuna.trial.TrialState,
 ) -> None:
-    study = optuna.create_study()
     dist = optuna.distributions.UniformDistribution(1.0, 100.0)
 
     # Prepare sampling result for later tests.
-    past_trials = [frozen_trial_factory(i, dist=dist) for i in range(1, 100)]
+    study = optuna.create_study()
+    for i in range(1, 100):
+        trial = frozen_trial_factory(i, dist=dist)
+        study._storage.create_new_trial(study._study_id, template_trial=trial)
```

Comment:
Looks very good to me to use `create_new_trial`. Thanks!

Reply:
We can do the same thing for other tests in this file but maybe in a separate PR.